src/Command/Anonymization/CleanCommand.php

```diff
@@ -64,32 +65,47 @@ protected function execute(InputInterface $input, OutputInterface $output): int
         }
 
         $connectionName = $input->getOption('connection') ?? $this->defaultConnectionName;
-        $anonymizator = $this->anonymizatorFactory->getOrCreate($connectionName);
+        $anonymizator = $this
+            ->anonymizatorFactory
+            ->getOrCreate($connectionName)
+            ->setOutput(new ConsoleOutput($io))
+        ;
 
-        $items = \iterator_to_array($anonymizator->clean(true));
+        $garbage = $anonymizator->collectGarbage();
 
-        if (!$items) {
-            $io->success("There is is no left-overs to clean, exiting.");
+        if (!$garbage) {
+            $io->success("There is no left-overs to clean, exiting.");
 
             return self::SUCCESS;
         }
 
         if (!$force) {
+            $tables = $others = [];
+
+            \array_walk($garbage, function ($item) use (&$tables, &$others) {
```

Comment:
Why not simply a foreach here ?

Reply:
You're right, this `\array_walk()` usage makes things pointlessly more complicated. I fixed it.